### PR TITLE
fix bus-redis worker to be able to connect to something other than redis on localhost

### PR DESCRIPTION
**Pull request by mod35**
Would be useful if the worker was able to connect to something other than the default of redis on localhost. It now respects the connection string.

tests pass:

image

### CODE REVIEW

File: packages/bus-redis/src/redis-transport.ts [hunk 1]
```diff
@@ -62,19 +63,21 @@ export class RedisMqTransport implements Transport<RedisMessage> {
 
   async initialize (): Promise<void> {
     this.logger.info('Initializing Redis transport')
-    this.connection = await this.connectionFactory()
+    this.queueConnection = await this.connectionFactory()
+    this.workerConnection = await this.connectionFactory()
```
Comment (adenhertog) [inline packages/bus-redis/src/redis-transport.ts:67]: Is there a need to create two connections to Redis for this, or can it use the one connection for both the queue & worker?

**Comment by adenhertog** [comment]
Quick one I might chat to you about, but it doesn't seem this implements pub sub per line 174 - it just pushes the message directly to the queue?

AFAIK redis supports pub sub https://redis.io/topics/pubsub, but bullmq hasn't added the docs yet https://docs.bullmq.io/patterns/sender-receiver ?

File: packages/bus-redis/src/redis-transport.ts (hunk 1 above)
Comment (mod35) [reply packages/bus-redis/src/redis-transport.ts:67]: Yes it can use the same - wasn't sure of the pros or cons of reusing the one vs scaling out.

**Comment by mod35** [comment]
I've also uncovered.some weird behaviour that I've asked about here
https://github.com/taskforcesh/bullmq/issues/681

**Comment by mod35** [comment]
> Quick one I might chat to you about, but it doesn't seem this implements pub sub per line 174 - it just pushes the message directly to the queue?
> 
> AFAIK redis supports pub sub https://redis.io/topics/pubsub, but bullmq hasn't added the docs yet https://docs.bullmq.io/patterns/sender-receiver ?

I think they are actively rewriting and so the docs are lacking

**Pull request closed without merging**